data_structures/delete_node_without_head.cpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * \file
+ * \brief [Delete a Node Without Head Pointer
+ * in Linked List](https://en.wikipedia.org/wiki/Linked_list#Singly_linked_list)
+ * @author [Anshu Saini](https://github.com/anshu189)
+ */
+
+#include <iostream>
+using namespace std;
+
+// Initializing a Linked List using Class Node
+class Node{
+public:
+    int data;
+    Node *next;
+    // Constructor for Linked List for a deafult value
+    Node(int data){
+        this -> data=data;
+        this -> next=NULL;
+    }
+};
+
+// Print Linked List Function
+void printList(Node* &head){
+    Node* temphead = head;
+    while(temphead != NULL){
+        cout << temphead -> data << " ";
+        temphead = temphead -> next;
+    }
+    cout<<endl;
+}
+
+// Appending elements in the Linked List 
+void insertattail(Node* &tail, int data){
+    Node *hue = new Node(data);
+    tail->next=hue;
+    tail=hue;
+}
+
+// Deleting a single Node without head Pointer in Linked List
+// Time Complexity: O(1)
+// Auxilliary Space: O(1)
+void delwohead(Node* del){
+    del->data=del->next->data;
+    del->next=del->next->next;
+}
+
+static void test( Node* head,Node* tail) {

```

Comment:
Preferably, do not use parameters in test functions. Instead, declare them inside the function.
```suggestion
/**
 * @brief Self-test implementations
 * @returns void
 */
static void test( Node* head,Node* tail) {
```